### ISSUE TITLE
S1643 - raises a problem when using compound assignment inside foreach on a list of objects.

### ISSUE THREAD

**Issue by twojnarowski** [SonarSource/sonar-dotnet#5819]
### Description

Rule S1643 is raised as a warning when I use a compound assignment for an object's property inside a foreach loop done for a list of objects.

### Repro steps

```
// downloading contracts with linq into a list of objects.
var contracts = ctx.Contracts.ToList();
foreach (var contract in contracts)
{
   contract.Notes += "Contract terminated!";
}
```

### Expected behavior

This should be correct.

### Actual behavior

Rule S1643 suggests I should use StringBuilder inside loops, but I can't use StringBuilder to edit one string parameter of each object.

### Known workarounds

None.

### Related information

* C#/VB.NET Plugins version -> SonarLint for Visual Studio 2022 6.6.0.49766
* Visual Studio version -> Visual Studio Community 2022 17.2.0
* MSBuild / dotnet version -> .NET Framework 4.8
* SonarScanner for .NET version (if used)
* Operating System -> Windows 10

**Comment by mary-georgiou-sonarsource** [comment]
Hello @twojnarowski and thank you for reporting this FP.
I will close this issue as is a duplicate of #5521.
